src/yz_kv.erl

```diff
@@ -395,6 +399,27 @@ cleanup(_, _) ->
     [].
 
 %% @private
+-spec cleanup({obj(), lp()}|bkey()) -> [{id, binary()}|{siblings, bkey()}].
+cleanup({Obj, LP}) ->
+    DocID = yz_doc:doc_id(Obj, ?INT_TO_BIN(LP)),
+    [{id, DocID}];
+cleanup(BKey) ->
+    [{siblings, BKey}].
+
+%% @private
+
+%% @doc Cleanup tombstone docs in object w/o siblings.
+-spec no_sibling_cleanup([doc()], obj(), lp()) -> []|[{id, doc()}].
+no_sibling_cleanup(Docs, Obj, LP) ->
+    lists:foldl(fun({doc, Fields}, Acc) ->
+                %% set to cleanup tombstone doc_ids if they exist
+                        case proplists:is_defined(tombstone, Fields) of
+                            true -> lists:append(Acc, cleanup({Obj, LP}));
```

Comment:
@jonmeredith the cleanup here could really also be the {siblings, BKey} case.. where it just deletes all previous vtags (TO:) for that key. we discussed doc_id originally though, so i went w/ that.